CRM/Case/BAO/Case.php

```diff
@@ -1994,37 +1994,48 @@ public static function getRelatedCaseIds($caseId, $excludeDeleted = TRUE) {
   }
 
   /**
-   * Retrieve case ids of all contacts related to given Organization contact.
+   * Retrieve contact ids of all contacts(with cases) related to given Organization contact.
    *
-   * @param int $orgId
+   * @param int $organizationId
    *
    * @return array
    */
-  public static function getOrganizationRelatedCaseIds($orgId) {
-    if(!$orgId) {
+  public static function getOrganizationRelatedCaseContactIds($organizationId, $countOnly=FALSE) {
+    if(!$organizationId) {
       return array();
     }
     $sql = "SELECT rel.contact_id_a as id FROM civicrm_case_contact AS cc
       INNER JOIN civicrm_relationship AS rel ON rel.contact_id_a = cc.contact_id
       INNER JOIN civicrm_relationship_type AS rtype ON rel.relationship_type_id = rtype.id
-      WHERE
-      'Organization' IN (rtype.contact_type_a, rtype.contact_type_b) AND
-      %1 = CASE
-      WHEN (rtype.contact_type_a = 'Organization')
-      THEN rel.contact_id_a
-      ELSE rel.contact_id_b
-      END ";
+      WHERE 'Organization' IN (rtype.contact_type_a, rtype.contact_type_b)
+      AND %1 = CASE
+        WHEN (rtype.contact_type_a = 'Organization') THEN rel.contact_id_a
+        ELSE rel.contact_id_b
+      END";
     $all = CRM_Core_DAO::executeQuery($sql, array(
-      1 => array($orgId, 'Integer'),
+      1 => array($organizationId, 'Integer'),
     ))->fetchAll();
     $ids = array();
+    if($countOnly) {
+      return count($all);
```

Comment:
We've had some problems in the past implementing counts like this (querying the database and then counting the number of elements of the array). For example, on ATL, where there are over 600K contacts. CiviCRM used to count the number of relationships for contacts like this, causing the relationships tab to break. It would be better if you use the flag to change the query and do the count at the database level.

eg:
```
SELECT COUNT(rel.contact_id_a) as total 
FROM civicrm_case_contact AS cc
INNER JOIN civicrm_relationship AS rel ON rel.contact_id_a = cc.contact_id
...
```